### PR TITLE
T209948: Fix UI and connection error

### PR DESCRIPTION
**Pull request by jan-jockusch**
The UI has a couple of stubborn checkboxes in the edit form, which should switch on or off depending on the actual settings.

This PR fixes that.

And, a new situation has popped up where the connection fails. This PR extends the classification for connection errors accordingly.

### CODE REVIEW

File: Products/ZPsycopgDA/dtml/add.dtml [hunk 1]
```diff
@@ -1,26 +1,23 @@
 <dtml-var manage_page_header>
 
 <dtml-var "manage_form_title(this(), _,
-           form_title='Add Z Psycopg 2 Database Connection',
-           help_product='ZPsycopgDA',
```
Comment (viktordick) [inline Products/ZPsycopgDA/dtml/add.dtml:5]: Why did you remove these?

File: Products/ZPsycopgDA/dtml/add.dtml [hunk 2]
```diff
@@ -62,14 +59,14 @@
     <div class="form-group row">
       <label for="check" class="form-label col-sm-4 col-md-3">Connect immediately</label>
       <div class="col-sm-5 col-md-1">
-        <input class="btn btn-primary" type="checkbox" id="check" name="check" value="YES" checked="YES" class="form-control"/>
+        <input class="btn btn-primary" type="checkbox" id="check" name="check" value="YES" checked="checked" />
```
Comment (viktordick) [inline Products/ZPsycopgDA/dtml/add.dtml]: Should the `form-control` class be added to the actual class attribute instead?

**Review by viktordick — COMMENTED** [review]
Some questions

File: Products/ZPsycopgDA/dtml/add.dtml (hunk 1 above)
Comment (jan-jockusch) [reply Products/ZPsycopgDA/dtml/add.dtml:5]: Oops, did I remove to much? help_product is useless as far as I can see, but the title wasn't supposed to go.

File: Products/ZPsycopgDA/dtml/add.dtml (hunk 2 above)
Comment (jan-jockusch) [reply Products/ZPsycopgDA/dtml/add.dtml]: form-control doesn't change a lot. Bootstrap says to use "form-check-input" to style check boxes, but that isn't used anywhere in the ZMI either.

Also, as above, not my idea originally.

File: Products/ZPsycopgDA/dtml/add.dtml (hunk 1 above)
Comment (jan-jockusch) [reply Products/ZPsycopgDA/dtml/add.dtml:5]: And the title stayed. So, this is actually OK, I believe.

Comment (jan-jockusch) [reply Products/ZPsycopgDA/dtml/add.dtml:5]: Still moved this into "draft" stage to have a little more time to learn about these things.

Comment (viktordick) [reply Products/ZPsycopgDA/dtml/add.dtml:5]: Do you have a pointer to the definition of `manage_form_title` so I can understand why it does not do much?

Comment (jan-jockusch) [reply Products/ZPsycopgDA/dtml/add.dtml:5]: The `manage_form_title` is actually a DTMLMethod with source code at `App/dtml/manage_form_title.dtml`.

There, if `help_product` and `help_topic` are both present, a help link is produced, by using a method from `HelpSys/HelpSys.py`.

The link uses the product name and the topic to build an URL which points at the structured text in the `help` directory.

The product "ZPsycopgDA" does not include a `help` directory. Therefore, this link is useless and the settings should be removed.

File: Products/ZPsycopgDA/dtml/add.dtml (hunk 2 above)
Comment (jan-jockusch) [reply Products/ZPsycopgDA/dtml/add.dtml]: Doing a code scan on Zope4, I see that "form-check-input" is used as a standard. Will add that.

The "btn btn-primary" styling is not used for checkboxes at all. That will go.

**Review by viktordick — APPROVED** [review]
LGTM
